### PR TITLE
AP_Scripting: add Lua bindings to check VTOL approach and land phases

### PR DESCRIPTION
**Pull request by CallanDaniel**
This PR adds Lua bindings for the follwing functions from quadplane.h :

- in_vtol_land_approach()
- in_vtol_land_final() 
- in_vtol_land_sequence() 
- in_vtol_land_poscontrol() 
- in_vtol_airbrake()

It also makes the functions public in order to expose them to Lua scritping. 

Tested in SITL with the example script running:

Screenshot 2024-07-11 114433

### CODE REVIEW

File: libraries/AP_Scripting/examples/vtol_land_status.lua [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    if quadplane:in_vtol_land_sequence() then
+     send_status_to_gcs()
+    end
+
+    -- Schedule the next run of this function in 1000 ms (1 Hz)
+    return update, 1000
+end
+
+-- Start the main loop
+return update, 1000
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/vtol_land_status.lua:51]:
```suggestion
return update, 1000

```

File: libraries/AP_Scripting/examples/vtol_land_status.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    local status = get_vtol_status()
+
+    -- Create the message to send to the GCS
+    local message = string.format("VTOL Status - Approach: %s, Final: %s, PosControl: %s, Airbrake: %s",
+                                  tostring(status.approach),
+                                  tostring(status.final),
+                                  tostring(status.poscontrol),
+                                  tostring(status.airbrake))
+-- Send the message to the GCS if in VTOL land sequence
+gcs:send_text(6, message)
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/vtol_land_status.lua:35]:
```suggestion
    -- Send the message to the GCS if in VTOL land sequence
    gcs:send_text(6, message)
```

File: libraries/AP_Scripting/examples/vtol_land_status.lua (hunk 1 above)
Comment (CallanDaniel) [reply libraries/AP_Scripting/examples/vtol_land_status.lua:51]: @IamPete1 did you just add an empty line at the end here?

Comment (Hwurzburg) [reply libraries/AP_Scripting/examples/vtol_land_status.lua:51]: he did
image
see the minus sign warning of no blank line at the end

Comment (CallanDaniel) [reply libraries/AP_Scripting/examples/vtol_land_status.lua:51]: Copy that @Hwurzburg appologies, read up on that now and makes complete sense. Thanks!

**Comment by CallanDaniel** [comment]
@Hwurzburg @IamPete1 I added the empty line, but it seems to dissapear when I commit. Appologies for the confusion, still learning.

**Comment by IamPete1** [comment]
> @Hwurzburg @IamPete1 I added the empty line, but it seems to dissapear when I commit. Appologies for the confusion, still learning.

It could be VS Code or git itself, sometimes it will not let you commit pure whitespace changes.

**Comment by tpwrules** [comment]
Bindings are kind of expensive, could you have an enum and one public function which checks and returns a specific property?

**Comment by CallanDaniel** [comment]
@tpwrules I don't see why not, this had actually crossed my mind when I was writing the example script. I'll do that. Thanks for the feedback!

**Comment by CallanDaniel** [comment]
@tpwrules something like this? Yet to do the bindings and a new example script - just wanted to check I'm going in the right direction.

**Pull request closed without merging**